tests/gnrc_tcp/tests/05-garbage-pkts.py

```diff
@@ -0,0 +1,146 @@
+
+
+@testfunc
+def test_short_header(child, src_if, src_ll, dst_if, dst_l2, dst_ll, dst_port):
+    # see https://github.com/RIOT-OS/RIOT/issues/12086
+    tcp_hdr = TCP(dport=dst_port, flags="S", sport=2342, seq=1, dataofs=6)
+    # shorten header
+    tcp_hdr = raw(tcp_hdr)[:-2]
+    sendp(Ether(dst=dst_l2) / IPv6(src=src_ll, dst=dst_ll) / tcp_hdr,
+          iface=src_if, verbose=0)
+
+    # let server command return to later check packet buffer
+    with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as sock:
+        sock.settimeout(child.timeout)
+        addr_info = socket.getaddrinfo(dst_ll + '%' + src_if, dst_port,
+                                       type=socket.SOCK_STREAM)
+        sock.connect(addr_info[0][-1])
+        child.expect_exact('gnrc_tcp_open_passive: returns 0')
+    verify_pktbuf_empty(child)
+
+
+@testfunc
+def test_send_ack_instead_of_syn(child, src_if, src_ll,
+                                 dst_if, dst_l2, dst_ll, dst_port):
+    # see https://github.com/RIOT-OS/RIOT/pull/12001
+    provided_data = base64.b64decode("rwsQf2pekYLaU+exUBBwgPDKAAA=")
+    tcp_hdr = TCP(provided_data)
+    assert provided_data == raw(tcp_hdr)
+    # set destination port to application specific port
+    tcp_hdr.dport = dst_port
+    sendp(Ether(dst=dst_l2) / IPv6(src=src_ll, dst=dst_ll) / tcp_hdr,
+          iface=src_if, verbose=0, count=1000)
+
+    # check if server actually still works
+    with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as sock:
+        sock.settimeout(child.timeout)
+        addr_info = socket.getaddrinfo(dst_ll + '%' + src_if, dst_port,
+                                       type=socket.SOCK_STREAM)
+        sock.connect(addr_info[0][-1])
+        child.expect_exact('gnrc_tcp_open_passive: returns 0')
+    verify_pktbuf_empty(child)
+
+
+@testfunc
+def test_option_parsing_term(child, src_if, src_ll,
+                             dst_if, dst_l2, dst_ll, dst_port):
+    # see https://github.com/RIOT-OS/RIOT/issues/12086
+    tcp_hdr = TCP(dport=dst_port, flags="S", sport=2342, seq=1, dataofs=6)
+    sendp(Ether(dst=dst_l2) / IPv6(src=src_ll, dst=dst_ll) / tcp_hdr /
+          b"\x50\x00\x00\x00", iface=src_if, verbose=0)
+
+    # check if server actually still works
+    with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as sock:
+        sock.settimeout(child.timeout)
+        addr_info = socket.getaddrinfo(dst_ll + '%' + src_if, dst_port,
+                                       type=socket.SOCK_STREAM)
+        sock.connect(addr_info[0][-1])
+        child.expect_exact('gnrc_tcp_open_passive: returns 0')
+    verify_pktbuf_empty(child)
+
+
+if __name__ == "__main__":
+    sudo_guard(uses_scapy=True)
+    script = sys.modules[__name__]
+    tests = [getattr(script, t) for t in script.__dict__
+             if type(getattr(script, t)).__name__ == "function" \
+             and t.startswith("test_")]
+    for test in tests:
+        res = run(test, timeout=1, echo=False)
```

Comment:
Please set default timeout to 7. I tried the test on 'nucleo-f401re' and 7 seconds was the threshold after the tests worked fine.

Reply:
I set it to 10 then.